libs/src/api/account.js

```diff
@@ -593,6 +595,43 @@ class Account extends Base {
       [AuthHeaders.SIGNATURE]: signature
     })
   }
+
+  /**
+   * Checks that the current user exists (i.e. has an account PDA) on SOL
+   * @returns {boolean} whether the user exists on SOL
+   */
+  async userHasAccountOnSol ({ userId, wallet } = { userId: null, wallet: null }) {
+    this.REQUIRES(Services.SOLANA_WEB3_MANAGER)
+
+    // If wallet or userId are not passed in, use the user loaded in libs
+    if (!wallet || !userId) {
+      const user = this.getCurrentUser()
+      wallet = user.wallet
+      userId = user.userId
+    }
+
+    // matches format for PDA derivation seed in SOL program
+    const userIdSeed = userId.toBuffer("le", 4)
+
+    const {
+      derivedAddress: userAccountPDA
+    } = await this.solanaWeb3Manager.findDerivedPair(
+      this.solanaWeb3Manager.audiusDataProgramId,
+      this.solanaWeb3Manager.audiusDataAdminStorageKeypairPublicKey,
+      userIdSeed
+    )
+
+    const account = await this.solanaWeb3Manager.fetchAccount(userAccountPDA)
+
+    if (!account) return false
+
+    const derivedEthWallet = await this.solanaWeb3Manager.deriveEthWalletFromAddress(account.ethAddress)
+
+    const userExistsOnChain = derivedEthWallet.toLowerCase() === wallet.toLowerCase()
+    const userUpgradedToSol = PublicKey.default.toString() !== account.authority.toString()
+
+    return userExistsOnChain && userUpgradedToSol
```

Comment:
One thing we want to be able to do is if the account is inited, but does not yet have an authority, then the user must claim the account to set it. It'd be nice if we could get that information from this method as well - or a separate method.

Reply:
thanks for the callout! will do

Reply:
gonna do this in the fast follow w/ testing fixes